test/ws/authorizationWebSocketHandlerSpec.ts

```diff
@@ -0,0 +1,260 @@
+import { WebSocketServer } from 'ws';
+import WebSocket from 'ws';
+import { AuthorizationWebSocketHandler } from '../../src/ws/authorization-websocket-handler';
+import { BookmarkManager } from '../../src/ws/bookmark-manager';
+import { InverseSpecificationEngine } from '../../src/ws/inverse-specification-engine';
+import { FactEnvelope, FactRecord, FactReference } from '../../src/storage';
+import { Specification } from '../../src/specification/specification';
+import { serializeGraph } from '../../src/http/serializer';
+
+// Mock invertSpecification to return both add and remove operations
+jest.mock('../../src/specification/inverse', () => {
+  const dummySpec: Specification = { given: [{ name: 'g', type: 'T' }], matches: [], projection: { type: 'composite', components: [] } };
+  return {
+    invertSpecification: jest.fn(() => ([
+      { inverseSpecification: dummySpec, operation: 'add', givenSubset: [], parentSubset: [], path: '', resultSubset: [] },
+      { inverseSpecification: dummySpec, operation: 'remove', givenSubset: [], parentSubset: [], path: '', resultSubset: [] },
+    ]))
+  };
+});
+
+jest.setTimeout(15000);
+
+describe('AuthorizationWebSocketHandler', () => {
+  let wss: WebSocketServer;
+
+  beforeAll(async () => {
```

Comment:
I fail to see how this WebSockets server is going to simulate a Jinaga replicator. I would expect to see an AuthenticationNoOp object backed by a memory store to simulate replicator behavior.